packages/vx-tooltip/src/hooks/useTooltip.ts

```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react';
+
+export type UseTooltipParams<TooltipData> = {
+  tooltipOpen: boolean;
+  tooltipLeft?: number;
+  tooltipTop?: number;
+  tooltipData?: TooltipData;
+  updateTooltip: (args: UpdateTooltipArgs<TooltipData>) => void;
+  showTooltip: (args: ShowTooltipArgs<TooltipData>) => void;
+  hideTooltip: () => void;
+};
+type UseTooltipState<TooltipData> = Pick<
+  UseTooltipParams<TooltipData>,
+  'tooltipOpen' | 'tooltipLeft' | 'tooltipTop' | 'tooltipData'
+>;
+type ShowTooltipArgs<TooltipData> = Omit<UseTooltipState<TooltipData>, 'tooltipOpen'>;
+type UpdateTooltipArgs<TooltipData> = UseTooltipState<TooltipData>;
+
+export default function useTooltip<TooltipData = {}>(): UseTooltipParams<TooltipData> {
+  const [{ tooltipOpen, tooltipLeft, tooltipTop, tooltipData }, setTooltipState] = useState<
```

Comment:
Since they're mostly updated together, I agree that a single `useState` with _all_ state is better than storing each variable separately in different `hook`s 👍